### DISCUSSION TITLE
how can I dynamically increase consumers

### DISCUSSION THREAD

**Discussion by ramprasadp** [apache/pulsar#19067] [General]
I am currently using kafka  , sorry no experience of pulsar
Our  production load is very peaky and distributed across multiple topics . 
Number of topics I restrict to "n" , usually 100 ,  because kafka does not support too many topics. And fixed 12 partitions 

The biggest challenge I see is that suddenly when the load ( number of events ) entering a single topic increases There is no way I can increase the number consumers for just that topic beyond 12 
The only way to scale seems to increase number of partitions which is not possible in runtime. Also the consumers are sometimes unevenly loaded and rebalancing kills the performance , for a good 5-10 minutes the consumers go starved when that h

Does Pulsar address these problems

**Comment by cbornet** [comment]
Yes, Pulsar addresses these problems by having more ways to consume the data.
If you use `Shared` or `KeyShared` subscriptions, you're not limited in the number of consumers that consume from the topic.
